up_client_socket/rust/utransport-socket/src/lib.rs

```diff
@@ -0,0 +1,520 @@
+    
+        let socket_sync = if let Ok(socket) = TcpStream::connect(DISPATCHER_ADDR) {
+            socket
+        } else {
+            error!("Error connecting sync socket");
+            return Err(UStatus::fail_with_code(UCode::INTERNAL, "Issue in connecting sync socket"));
+        };
+    
+       let listener_map = Arc::new(Mutex::new(HashMap::new())) ;
+
+       let socket_clone = match socket_sync.try_clone() {
+        Ok(socket) => socket,
+        Err(err) => {
+      
+            error!("Issue in cloning sync socket: {}", err);
+            return Err(UStatus::fail_with_code(UCode::INTERNAL, "Issue in cloning socket_sync"));
+        }
+    };
+       let mut transport_socket = UTransportSocket {
+        socket_sync:socket_clone,
+        listener_map: listener_map.clone(),
+    };
+        match transport_socket.socket_init(){
+            Ok(_) => {
+                // The function call succeeded
+                debug!("socket trasport initilized successfully");
+            }
+            Err(err) => {
+                // The function call failed with an error
+                error!("socket trasport initilized failed: {}", err);
+                return Err(UStatus::fail_with_code(UCode::INTERNAL, "Issue in cloning socket_sync"));
+            }
+        }
+        debug!("UTransportSocket initilization successful");
+       
+        Ok( UTransportSocket {
+             socket_sync,
+             listener_map,
+         })
+     }
+ 
+     /// # Panics
+     ///
+     /// Will panic if failed to parse message
+      fn socket_init(&mut self) -> Result<(), UStatus>{
+        let socket_clone = match self.socket_sync.try_clone() {
+            Ok(socket) => socket,
+            Err(err) => {
+          
+                error!("Issue in cloning sync socket: {}", err);
+                return Err(UStatus::fail_with_code(UCode::INTERNAL, "Issue in cloning socket_sync"));
+            }
+        };
+    
+    let listener_map_clone = self.listener_map.clone();
+    
+    let mut self_copy = Self {
+        socket_sync: socket_clone,
+        listener_map: listener_map_clone,
+    };
+    task::spawn(async move { self_copy.dispatcher_listener().await });
+   
+    
+    Ok(())
+}
+    async  fn dispatcher_listener(&mut self) {
+        debug!("started listener for dispatcher");
+         loop {
+             // Receive data from the socket
+             let mut buffer: [u8; BYTES_MSG_LENGTH] = [0; BYTES_MSG_LENGTH];
+             let bytes_read = match self.socket_sync.read(&mut buffer) {
+                 Ok(bytes_read) => bytes_read,
+                 Err(e) => {
+                     debug!("Socket error: {}", e);
+                     break;
+                 }
+             };
+ 
+             // Check if no data is received
+             if bytes_read == 0 {
+                 continue;
+             }
+ 
+             let umessage_result = UMessage::parse_from_bytes(&buffer[..bytes_read]);
```

Comment:
It's weird, the other files don't seem to have this issue, but this one's formatting is kinda off. Please check the indenting / formatting.